src/main/webapp/creatives/maps/simid_maps.js

```diff
@@ -1,8 +1,44 @@
 /**
  * A sample SIMID ad that shows a map of nearby locations
  */
+const AdParamKeys = {
+  BUTTON_LABEL: 'buttonLabel',
+  SEARCH_QUERY: 'searchQuery',
+  MARKER: 'marker',
+}
+const DEFAULT_BUTTON_LABEL = "Location";
+
 class SimidMapCreative extends BaseSimidCreative {
   constructor() {
     super();
   }
+
+  /** @override */
+  onStart(eventData){
+    super.onStart(eventData);
+    const adParams = this.creativeData.adParameters;
```

Comment:
Call JSON.parse here and save that as the variable, so you don't have to call it again each time you use it

Reply:
sounds good